mig/shared/settings.py

```diff
@@ -62,26 +63,29 @@ def parse_and_save_pickle(source, destination, keywords, client_id,
     client_dir = client_id_dir(client_id)
     result = parse(source, strip_space, strip_comments)
 
-    (status, parsemsg) = check_types(result, keywords, configuration)
+    # NOTE: check_types also fills parsed results into the passed keywords dict
+    parsed = copy.deepcopy(keywords)
```

Comment:
I don't quite understand what the deepcopy is achieving here unless check_types is modifying the structure in-place!?

Reply:
Alas, and to my horror that is exactly what it does :-( as I tried to highlight in the comment.
This is ancient code used in a number of modules so while it _should_ be fixed - at least with a renaming of that function, I did not deem it suitable for this same PR.  I'll add a TODO about it during my follow-up to your review and mark the PR with the follow-up tag.